test/dynamo/test_install_params_as_graph_attr.py

```diff
@@ -116,9 +135,91 @@ def forward(self, x) -> torch.Tensor:
         input1 = torch.randn((1, 5))
-        self.check_num_inputs_and_equality(net, input1, 1 + 2 * (kDepth + 1))
+        self.check_num_inputs_and_equality(net, 1 + 2 * (kDepth + 1), 1, (input1,))
+
+    def test_simple_batchnorm(self) -> None:
+        net = torch.nn.BatchNorm2d(3)
+        tensor = torch.randn((1, 3, 3, 3))
+        # BatchNorm2d has 2 params, and 3 buffers
+        self.check_num_inputs_and_equality(net, 6, 1, (tensor,))
+
+    def test_nets_as_input(self) -> None:
+        """
+        Tests when the nn.Module is an input to the fn we are optimizing
+
+        In this case, we should treat it as regular input, which means we
+        can lift parameters/buffers, but should not install them
+        """
+        # Test nn model as input
+        net = SimpleLinearModule()
+        net2 = SimpleLinearModule()
+        x = torch.randn(1, 5)
+
+        def test_fn(x, net):
+            return net(x)
+
+        # When nn is in input, we don't install the params
+        self.check_num_inputs_and_equality(test_fn, 3, 3, (x, net))
+
+        def test_fn2(x, net, net2):
+            return net(x) + net2(x)
+
+        self.check_num_inputs_and_equality(test_fn2, 5, 5, (x, net, net2))
+
+        def test_fn3(x, net):
+            return net(x) + net2(x)
+
+        # In case of local scope (net2 here), we can install
+        self.check_num_inputs_and_equality(test_fn3, 5, 3, (x, net))
 
-    # TODO[@lucaskabela]: Test nontrivial such as resnet, ffn, or transformer
+        def test_fn_list(x, nets):
+            return sum([net(x) for net in nets])
+
+        self.check_num_inputs_and_equality(test_fn_list, 5, 5, (x, [net, net2]))
+
+    def test_resnet_structure(self) -> None:
+        class ResBlock(torch.nn.Module):
+            def __init__(self, in_, out_):
+                super().__init__()
+                self.conv1 = torch.nn.Sequential(
+                    torch.nn.Conv2d(in_, out_, kernel_size=3, padding=1),
+                    torch.nn.BatchNorm2d(out_),
+                    torch.nn.ReLU(),
+                )
+                self.conv2 = torch.nn.Sequential(
+                    torch.nn.Conv2d(out_, out_, kernel_size=3, padding=1),
+                    torch.nn.BatchNorm2d(out_),
+                )
+                self.activation = torch.nn.ReLU()
+
+            def forward(self, x):
+                skip = x
+                out = self.conv1(x)
+                out = self.conv2(out)
+                out += skip
+                out = self.activation(out)
+                return out
+
+        net = ResBlock(3, 3)
+        tensor = torch.randn(1, 3, 3, 3)
+        # Conv2d has 2 params, BatchNorm2d has 3 buffers + 2 params, and Relu has 0 params
+        # So expected = 2 + 5 + 5 + 2 = 14 + 1 for input
+        self.check_num_inputs_and_equality(net, 15, 1, (tensor,))
+
+    def test_transformer(self) -> None:
+        # NOTE: Transformer fails the equality test, so skip that check
```

Comment:
I am wondering if this is expected? I.e running:

```
torch.compile(transformer)(src, tgt) != transformer(src, tgt)
```

so this is why I do not just use the helper fn here